### PR TITLE
feat: create new note, update note, add tags, change theme colors.

### PR DESCRIPTION
**Pull request by tushartiwari7**
Feat: Create new Note
Feat: Update Note with title,content,tags and themeColors
Feat: AutoSave, Notes are Synced to the backend and don't need to be saved manually.

image

### CODE REVIEW

File: src/services/needAxios.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
     const { data, status } = await axios({
       method,
       url,
       data: body,
       headers: {
-        authorization: localStorage.getItem("token") ?? "",
+        authorization: token ?? "",
       },
     });
     return { data, status };
```
Comment (munjalSaksham71) [inline src/services/needAxios.js:14]: This is really cool!

File: src/services/note.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { needAxios } from "./needAxios";
+
+export const createNew = async () => {
+  const { data, status } = await needAxios("POST", "/api/notes", {
```
Comment (munjalSaksham71) [inline src/services/note.js:4]: Awesome!!

**Review by munjalSaksham71 — APPROVED** [review]
The code looks good to me and understandable!
Liked that you have using ref for current state and obviously debouncing. 
Please try to add the UI's screenshot or deploy preview.

**Review by Namrata-J — APPROVED** [review]
Naming used are explanatory which makes the code more understandable, and also the code is well structured!!

**Review by Mohitkumar33 — APPROVED** [review]
nicely structured and clean code

**Pull request merged**